conan/tools/b2/b2deps.py

```diff
@@ -0,0 +1,284 @@
+from conan.internal import check_duplicated_generator
+from conan.tools.b2.util import *
+from conans.errors import ConanException
+from conans.util.files import save, chdir
+from conans.paths import get_conan_user_home
+from hashlib import md5
+
+
+class B2Deps(object):
+    """
+    B2Deps generates files that are automatically loaded by the B2 build system.
+    The files define localized subprojects and targets for dependencies.
+    """
+
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+        self._conanhome = get_conan_user_home()
```

Comment:
I am still not sure about this. Generators also must work the same when a package is in ``editable`` mode for example, so the package is not in the cache at all, not in the CONAN_HOME. The same should happen for "deployed" artifacts, like https://blog.conan.io/2023/05/23/Conan-agnostic-deploy-dependencies.html, where the generated files will be used without Conan, and should be relocatable.

> It's used to figure out host independent paths instead of the absolute paths that normally end up in generator files.

This is another concern that should be a bit more orthogonal. I think it is better to have first the "simple" approach, in which paths are absolute paths to the cache, and work from there, based on the needs, than adding this that might have other unexpected rough edges

Reply:
> Thing is where b2 deps puts the generated files it's expected for them to be stable and likely included in source control.

I am afraid this is not the design of all Conan generators. They are not expected to be stable, but they are located in most cases inside the "build" folder (the ``generators`` folder inside ``build`` folder) and intended to be ``.gitignored`` and cleaned regularly. The reason for this is that these files can and will change for every single configuration change (architecture, shared, build_type), and not absolutely every binary variant can be captured in a file path of a generator.

I still think that this must be simplified and follow the rest of the generators design, allowing absolute paths in the generated files and managing them as non stable and not in version control.